cedar-policy-validator/src/human_schema/ast.rs

```diff
@@ -0,0 +1,441 @@
+use std::iter::once;
+
+use cedar_policy_core::{
+    ast::Id,
+    parser::{Loc, Node},
+};
+use itertools::Itertools;
+use nonempty::NonEmpty;
+use smol_str::SmolStr;
+
+use crate::SchemaTypeVariant;
+
+const IPADDR_EXTENSION: &str = "ipaddr";
+const DECIMAL_EXTENSION: &str = "decimal";
+pub const EXTENSIONS: [&str; 2] = [IPADDR_EXTENSION, DECIMAL_EXTENSION];
+pub const BUILTIN_TYPES: [&str; 3] = ["Long", "String", "Bool"];
+
+pub(super) const CEDAR_NAMESPACE: &str = "__cedar";
+
+pub type Schema = Vec<Node<Namespace>>;
+
+/// A path is a non empty list of identifiers that forms a namespace + type
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct Path(Node<PathInternal>);
+impl Path {
+    /// Create a [`Path`] with a single entry
+    pub fn single(basename: Id, loc: Loc) -> Self {
+        Self(Node::with_source_loc(
+            PathInternal {
+                basename,
+                namespace: vec![],
+            },
+            loc,
+        ))
+    }
+
+    /// Create [`Path`] with a head and an iterator. Most significant name first.
```

Comment:
I read "Most significant" as being leftmost, but then this function sticks it in `basename` which I interpret as rightmost?  but maybe I just interpret one of those terms weirdly / different from everyone else

Reply:
oh this means most significant _inside the iterator_ first.  the parameter name `basename` is sufficient documentation for what that represents